### DISCUSSION TITLE
Change location of the edge label to simulate movement

### DISCUSSION THREAD

**Discussion by Sharaf0** [xyflow/xyflow#2528] [Q&A]
Hi there,
Is there a way to change location of the label of the edge? I want to simulate data flowing between nodes, something like movements in the attached picture.
![pic](https://user-images.githubusercontent.com/3614680/198855792-55bfbdb7-eb71-4e34-9a3f-2823d1af9a3c.gif)

**Comment by moklick** [comment]
That's an interesting thing to do :) It's not built-in but possible to do with a custom edge implementation by using the [`getPointAtLength` function](https://developer.mozilla.org/en-US/docs/Web/API/SVGGeometryElement/getPointAtLength) and then animate the icon along the path.
